application/src/main/java/org/togetherjava/tjbot/commands/help/HelpSystemHelper.java

```diff
@@ -239,6 +250,50 @@ List<ThreadChannel> getActiveThreadsIn(TextChannel channel) {
             .toList();
     }
 
+    void scheduleUncategorizedAdviceCheck(long threadChannelId, long authorId) {
+        SERVICE.schedule(() -> {
+            try {
+                executeUncategorizedAdviceCheck(threadChannelId, authorId);
+            } catch (Exception e) {
+                logger.warn(
+                        "Unknown error during an uncategorized advice check on thread {} by author {}.",
+                        threadChannelId, authorId, e);
+            }
+        }, SEND_UNCATEGORIZED_ADVICE_AFTER_MINUTES, TimeUnit.MINUTES);
+    }
+
+    private void executeUncategorizedAdviceCheck(long threadChannelId, long authorId) {
+        logger.debug("Executing uncategorized advice check for thread {} by author {}.",
+                threadChannelId, authorId);
+        jda.retrieveUserById(authorId).flatMap(author -> {
+            ThreadChannel threadChannel = jda.getThreadChannelById(threadChannelId);
+            if (threadChannel == null) {
+                logger.debug(
+                        "Channel for uncategorized advice check seems to be deleted (thread {} by author {}).",
+                        threadChannelId, authorId);
+                return new CompletedRestAction<>(jda, null);
+            }
+
+            Optional<String> category = getCategoryOfChannel(threadChannel);
+            if (category.isPresent()) {
+                logger.debug(
+                        "Channel for uncategorized advice check seems to have a category now (thread {} by author {}).",
+                        threadChannelId, authorId);
+                return new CompletedRestAction<>(jda, null);
+            }
+
+            // Still no category, send advice
+            MessageEmbed embed = HelpSystemHelper.embedWith(
+                    """
+                            Hey there 👋 You have to select a category for your help thread, otherwise nobody can see your question.
+                            Please use the `/change-help-category` slash-command and pick what fits best, thanks 🙂
+                            """);
```

Comment:
Can't we make this a field? At the very least make the string a field.

Reply:
Why? What would be the advantage? Its already very isolated with it being a single argument to the factory.